### PR TITLE
Allow update initial failover version and version increment

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Allow update initial failover version and version increment if not enable global namespace

**Why?**
Give it a chance to update the failover version and version increment.

**How did you test it?**
Tested locally

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: temporal/fx.go [hunk 1]
```diff
@@ -593,25 +593,12 @@ func ApplyClusterMetadataConfigProvider(
...
-		}
-		if resp.FailoverVersionIncrement != clusterData.FailoverVersionIncrement {
-			logger.Warn(
-				mismatchLogMessage,
-				tag.Key("clusterMetadata.FailoverVersionIncrement"),
-				tag.IgnoredValue(clusterData.FailoverVersionIncrement),
-				tag.Value(resp.FailoverVersionIncrement))
-			config.ClusterMetadata.FailoverVersionIncrement = resp.FailoverVersionIncrement
-		}
 		if resp.IsGlobalNamespaceEnabled != clusterData.EnableGlobalNamespace {
```
Comment (yiminc) [inline temporal/fx.go:596]: I'm not sure it should be allowed to downgrade for IsGlobalNamespaceEnabled from true to false. What is the implication?

File: common/persistence/clusterMetadataStore.go [hunk 2]
```diff
@@ -206,9 +206,16 @@ func (m *clusterMetadataManagerImpl) convertInternalGetClusterMetadataResponse(
 
 // immutableFieldsChanged returns true if any of immutable fields changed.
 func immutableFieldsChanged(old persistencespb.ClusterMetadata, cur persistencespb.ClusterMetadata) bool {
-	return (old.ClusterName != "" && old.ClusterName != cur.ClusterName) ||
+	if (old.ClusterName != "" && old.ClusterName != cur.ClusterName) ||
```
Comment (yiminc) [inline common/persistence/clusterMetadataStore.go:209]: what about clusterName? Should you allow clusterName change if globNamespace is not enabled yet on current cluster.

Comment (yycptt) [reply common/persistence/clusterMetadataStore.go:209]: Cluster name is also stored in namespace table so probably not a good idea to change it?

Comment (yux0) [reply common/persistence/clusterMetadataStore.go:209]: Yes. This is the tricky one. We need some offline discussion.

File: temporal/fx.go (hunk 1 above)
Comment (yux0) [reply temporal/fx.go]: Updated

File: common/persistence/clusterMetadataStore.go (hunk 2 above)
Comment (yiminc) [reply common/persistence/clusterMetadataStore.go:209]: 👍

**Review by yiminc — APPROVED** [review]

**Pull request merged**